mfs/ops.go

```diff
@@ -46,21 +34,42 @@ func Mv(r *Root, src, dst string) error {
 		return err
 	}
 
-	fsn, err := dstDir.Child(filename)
-	if err == nil {
-		switch n := fsn.(type) {
-		case *File:
-			_ = dstDir.Unlink(filename)
-		case *Directory:
-			dstDir = n
-		default:
-			return fmt.Errorf("unexpected type at path: %s", dst)
+	var dstDirStr string
+	var dstDir *Directory
+	if dst[len(dst)-1] == '/' {
+		dstDirStr = dst
+		dstDir, err = lookupDir(r, dstDirStr)
+		if err != nil {
+			return err
+		}
+		_, err = dstDir.Child(srcFname)
+		if err == nil {
+			return os.ErrExist
+		}
+
+	} else {
+		dstDirStr, dirFileName := gopath.Split(dst)
+		dstDir, err = lookupDir(r, dstDirStr)
+		if err != nil {
+			return err
+		}
+
+		fsn, err := dstDir.Child(dirFileName)
+		if err == nil {
+			switch n := fsn.(type) {
+			case *File:
+				return fmt.Errorf("%s : is not a directory", dst)
```

Comment:
Again, we should overwrite in this case (if the source is also a file).

Reply:
fixed